### PR TITLE
Organisms eat food 

### PR DESCRIPTION
**Pull request by insoochoo**
not sure waht is wrong. please check it out

### CODE REVIEW

File: modules/organism.js [hunk 1]
```diff
@@ -34,15 +35,45 @@ Organism.prototype = {
...
 
     this.checkBoundaries(land);
   },
 
-  findFriends: function(population, range)
+  // follow until inside target's radius
+  follow: function(target, radius)
   {
-    var friends = [];
+    var dest = target.copy().sub(this.location);
```
Comment (bumjung) [inline modules/organism.js]: can we have a more descriptive var name?

Comment (insoochoo) [inline modules/organism.js]: dest = destination?

Comment (insoochoo) [inline modules/organism.js]: oh also, this code isnt for submit

**Comment by insoochoo** [comment]
this needs a revamp because something fundamental is wrong

File: modules/organism.js [hunk 2]
```diff
@@ -34,15 +35,45 @@ Organism.prototype = {
...
+    var distance = dest.mag();
+
+    if (distance > radius)
+    {
+      dest.setMag(this.maxSpeed);
```
Comment (bumjung) [inline modules/organism.js:90]: organism is going to travel max speed to food?

Comment (insoochoo) [inline modules/organism.js:90]: we need to calculate that force too.

**Comment by insoochoo** [comment]
still WIP

File: modules/organism.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ Organism.prototype = {
         var food = this.nearByFood[i];
         if (!food.depleted)
         {
-          this.follow(food.location, food.radius, food.mass * 10);
+          this.follow(food, food.radius, food.energy/(Constant.ENERGY*2));
```
Comment (bumjung) [inline modules/organism.js]: we should add
`food.mass = food.energy / Constant.ENERGY`
in Food.js, and use
`food.mass / 2`

File: modules/organism.js [hunk 4]
```diff
@@ -161,12 +165,15 @@ Organism.prototype = {
     ctx.fillStyle = "black";
     ctx.strokeStyle = "black";
     ctx.beginPath();
-    ctx.arc(this.location.x, this.location.y, this.mass * 100000 * Constant.RADIUS_RATIO, 0, 2 * Math.PI, false);
+    ctx.arc(this.location.x, this.location.y, this.mass * Constant.ENERGY * Constant.RADIUS_RATIO, 0, 2 * Math.PI, false);
     ctx.stroke();
     ctx.fill();
 
-    this.drawVision(ctx);
```
Comment (bumjung) [inline modules/organism.js]: did you rebase?

**Comment by insoochoo** [comment]
just rebased

File: modules/organism.js [hunk 5]
```diff
@@ -34,15 +36,70 @@ Organism.prototype = {
...
+    this.nearByFood = this.findVisibleFood(land.food);
+
+    if (this.nearByFood.length)
+    {
+      for (var i in this.nearByFood)
+      {
+        var food = this.nearByFood[i];
+        if (!food.depleted)
+        {
+          this.follow(food, food.radius, food.energy/(Constant.ENERGY*2));
```
Comment (bumjung) [inline modules/organism.js]: I think `food.mass / 2` is better

**Comment by bumjung** [comment]
:+1: :cactus:

**Pull request merged**